siriuspy/siriuspy/currinfo/lifetime/main.py

```diff
@@ -0,0 +1,166 @@
+            self.run_callbacks('MinIntvlBtwSpl-RB', self._min_intvl_btw_spl)
+            status = True
+        elif reason == 'LtFitMode-Sel':
+            self._mode = value
+            self.run_callbacks('LtFitMode-Sts', value)
+            status = True
+        elif reason == 'CurrOffset-SP':
+            self._current_offset = value
+            self.run_callbacks('CurrOffset-RB', value)
+            status = True
+        return status
+
+    def _update_min_intvl_btw_spl(self, value):
+        if value < 0:
+            value = 0
+        self._min_intvl_btw_spl = value
+
+    def _update_splintvl(self, value):
+        self._sampling_interval = value
+
+    def _callback_get_storedebeam(self, value, **kws):
+        self._is_stored = value
+
+    def _callback_calclifetime(self, pvname, timestamp, **kws):
+        # check DCCT StoredEBeam PV
+        if not self._is_stored:
+            return
+
+        is_bpm = 'BPM' in pvname
+        lt_type = 'BPM' if is_bpm else ''
+        lt_name = '_lifetime'+('_bpm' if is_bpm else '')
+        buffer_dt = self._bpmsum_buffer if is_bpm else self._current_buffer
+        buffer_lt = self._lifetime_bpm_buffer \
+            if is_bpm else self._lifetime_dcct_buffer
+
+        # try to add a new point to buffer
+        acquireflag = buffer_dt.acquire()
+        if not acquireflag:
+            return
+
+        # check min number of points in buffer to calculate lifetime
+        [ts_abs_dqorg, val_abs_dqorg] = buffer_dt.get_serie(time_absolute=True)
+        buffer_dt_aux = _SiriusPVTimeSerie(
+            pv=None,
+            timestamp_init_data=ts_abs_dqorg,
+            value_init_data=val_abs_dqorg)
+        buffer_dt_aux.time_min_interval = self._min_intvl_btw_spl
+        buffer_dt_aux.time_window = self._sampling_interval
+        ts_dq, val_dq = buffer_dt_aux.get_serie()
+        ts_dq = _np.array(ts_dq)
+        val_dq = _np.array(val_dq)
+        if not len(ts_dq):
+            setattr(self, lt_name, 0)
+        else:
+            val_dq -= self._current_offset
+            if len(val_dq) > 100:
+                fit = 'exp' if self._mode == _Const.Fit.Exponential else 'lin'
+                lt = self._least_squares_fit(ts_dq, val_dq, fit=fit)
+                setattr(self, lt_name, lt)
+                buffer_lt.append(lt)
+
+        # update pvs
+        self.run_callbacks('BufferValue'+lt_type+'-Mon', val_dq)
+        self.run_callbacks('BufferTimestamp'+lt_type+'-Mon', ts_dq)
+        self.run_callbacks('Lifetime'+lt_type+'-Mon', getattr(self, lt_name))
+        self.run_callbacks('BuffSize'+lt_type+'-Mon', len(val_dq))
+        self.run_callbacks('BuffSizeTot'+lt_type+'-Mon', len(val_abs_dqorg))
+
+    @staticmethod
+    def _least_squares_fit(timestamp, value, fit='exp'):
+        if fit == 'exp':
+            try:
+                value = _np.log(value)
+            except Exception:
+                return 0.0
+        n = len(timestamp)
```

Comment:
@anacso17 , we are discussing how to deal with pylint pedantic complaints on three-letter variable name rule. We would like to rely on pylint to improve the code but having it listing these type of errors is a noiance. While we do not implement a definite solution maybe you can change the variable names. One simple solution is to use a prefix underscore.

See and contribute to the discussion I created for us in 
https://gitter.im/lnls-fac/community#

Reply:
I renamed the variable!